node/sequelize/models/index.js

```diff
@@ -1,29 +1,216 @@
+const FAILED_HOST_RECONNECT_DELAY_SECS = process.env.PGFAILEDHOSTRECONNECTDELAYSECS ? Number(process.env.PGFAILEDHOSTRECONNECTDELAYSECS) : 5;
+
+// Utility function to set environment variables if not already set
+// The @yugabytedb/pg driver reads these directly from process.env
+const ensureEnv = (key, value) => {
+  if (
+    typeof value !== 'undefined' &&
+    value !== null &&
+    value !== '' &&
+    !process.env[key]
+  ) {
+    process.env[key] = String(value);
+  }
+};
+
+// Set environment variables for @yugabytedb/pg driver
+ensureEnv('PGLOADBALANCE', READ_LOAD_BALANCE_MODE);
+ensureEnv('PGTOPOLOGYKEYS', TOPOLOGY_KEYS);
+ensureEnv(
+  'PGFALLBACKTOTOPOLOGYKEYSONLY', 
+  FALLBACK_TO_TOPOLOGY_KEYS_ONLY ? 'true' : undefined,
+);
+ensureEnv('PGYBSERVERSREFRESHINTERVAL', SERVER_REFRESH_INTERVAL);
+ensureEnv(
+  'PGFAILEDHOSTRECONNECTDELAYSECS',
+  FAILED_HOST_RECONNECT_DELAY_SECS,
+);
+
+function createSmartSequelizeInstance() {
+  const [primaryHost] = HOSTS;
+
+  const username = process.env.PGUSER || baseConfig.username || 'yugabyte';
+  const password = process.env.PGPASSWORD || (baseConfig.password && baseConfig.password !== '' ? baseConfig.password : 'yugabyte');
+  const database = process.env.PGDATABASE || baseConfig.database || 'ysql_sequelize';
+
+  const logLevel = (process.env.LOG_LEVEL || '').toLowerCase();
+  
+  const loggingType =
+    logLevel === 'silly'
+      ? console.log
+      : false;
+
+  // Use YugabyteDB smart driver for automatic load balancing
+  // Driver reads config from process.env/environment variables (set via ensureEnv) and discovers all nodes
+  return new Sequelize(database, username, password, {
+    host: primaryHost.host,
+    port: primaryHost.port,
+    dialect: 'postgres',
+    logging: loggingType,
+    pool: {
+      max: 10,
+      min: 2,
+      idle: 10000,
+      acquire: 30000,
+    },
+    dialectOptions: {
+      statement_timeout: 60000,
+    },
+  });
+}
+
+// Alternative method: Using Connection String
+function createSequelizeWithConnectionString() {
+  const username = process.env.PGUSER || baseConfig.username || 'yugabyte';
+  const password = process.env.PGPASSWORD || (baseConfig.password && baseConfig.password !== '' ? baseConfig.password : 'yugabyte');
+  const database = process.env.PGDATABASE || baseConfig.database || 'ysql_sequelize';
+  
+  // Use first host for simple connection
+  const { host, port } = HOSTS[0];
+  
+  // Build connection string with load balance parameters
+  const connectionString = 
+    `postgres://${encodeURIComponent(username)}:${encodeURIComponent(password)}@${host}:${port}/${database}` +
+    `?loadBalance=${READ_LOAD_BALANCE_MODE}` +
+    (TOPOLOGY_KEYS ? `&topologyKeys=${encodeURIComponent(TOPOLOGY_KEYS)}` : '') +
+    `&ybServersRefreshInterval=${SERVER_REFRESH_INTERVAL}` +
+    `&failedHostReconnectDelaySecs=${FAILED_HOST_RECONNECT_DELAY_SECS}` +
+    (FALLBACK_TO_TOPOLOGY_KEYS_ONLY ? `&fallbackToTopologyKeysOnly=true` : '');
```

Comment:
Let's not use all that we offer. Users can find that from the smart drivers page.
```suggestion
  const connectionString = 
    `postgres://${encodeURIComponent(username)}:${encodeURIComponent(password)}@${host}:${port}/${database}` +
    `?loadBalance=${READ_LOAD_BALANCE_MODE}` +
    (TOPOLOGY_KEYS ? `&topologyKeys=${encodeURIComponent(TOPOLOGY_KEYS)}` : '') +
    `&ybServersRefreshInterval=${SERVER_REFRESH_INTERVAL}` : '');
```